app/models/service_template/copy.rb

```diff
@@ -0,0 +1,18 @@
+module ServiceTemplate::Copy
+  extend ActiveSupport::Concern
+
+  def copy(new_name = "Copy of " + name + Time.zone.now.to_s)
+    if template_valid? && type != 'ServiceTemplateAnsiblePlaybook'
+      ActiveRecord::Base.transaction do
+        dup.tap do |template|
+          template.update_attributes(:name => new_name, :guid => nil, :display => false)
+          service_resources.each do |sr|
+            resource = sr.resource.respond_to?(:service_template_resource_copy) ? sr.resource.service_template_resource_copy : sr.resource
+            template.add_resource(resource)
```

Comment:
The `service_resource` instance is a join-table that carries additional metadata (like scaling and start/stop actions) that needs to be applied to the new instance as well.  The `add_resource` method accepts an optional parameter of `service_resource` and will copy the values.